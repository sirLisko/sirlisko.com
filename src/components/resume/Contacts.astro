<article class="section contacts">
  <h1>Contacts</h1>
  <p>
    Personal website:{" "}
    <a href="https://sirlisko.com">https://sirlisko.com</a>
  </p>
  <p>
    Github:{" "}
    <a href="https://github.com/sirlisko">https://github.com/sirlisko</a>
  </p>
  <p>
    LinkedIn:{" "}
    <a href="https://www.linkedin.com/in/lucalischetti/">
      https://www.linkedin.com/in/lucalischetti/
    </a>
  </p>
  <p>
    Email:{" "}
    <a href="mailto:luca@sirlisko.com">luca@sirlisko.com</a>.
  </p>
  <p class="alt-hidden">
    A print-friendly version of this:{" "}
    <a href="https://sirlisko.com/cv/alt.pdf">
      https://sirlisko.com/cv/alt.pdf
    </a>
  </p>
  <p class="alt-visible">
    A live version of this resume:{" "}
    <a href="https://sirlisko.com/resume">https://sirlisko.com/resume</a>
  </p>
</article>
<style is:global>
  .alt-visible {
    display: none;
  }
  .alt .alt-visible {
    display: block;
  }
  .alt .alt-hidden {
    display: none;
  }
</style>
