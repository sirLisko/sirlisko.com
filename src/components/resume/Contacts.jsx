import React from "react";
import PropTypes from "prop-types";

const Contacts = ({ isAltVersion }) => (
  <article className="section contacts">
    <h1>Contacts</h1>
    <p>
      My personal website{" "}
      <a href="https://sirlisko.com">https://sirlisko.com</a>
    </p>
    <p>
      My Github{" "}
      <a href="https://github.com/sirlisko">https://github.com/sirlisko</a>
    </p>
    <p>
      My LinkedIn{" "}
      <a href="https://www.linkedin.com/in/lucalischetti/">
        https://www.linkedin.com/in/lucalischetti/
      </a>
    </p>
    <p>
      Drop me an email at{" "}
      <a href="mailto:luca@sirlisko.com">luca@sirlisko.com</a>.
    </p>
    {!isAltVersion ? (
      <p>
        A print-friendly version of this:{" "}
        <a href="https://sirlisko.com/cv/alt.pdf">
          https://sirlisko.com/cv/alt.pdf
        </a>
      </p>
    ) : (
      <p>
        A live version of this resume:{" "}
        <a href="https://sirlisko.com/resume">https://sirlisko.com/resume</a>
      </p>
    )}
  </article>
);

Contacts.propTypes = {
  isAltVersion: PropTypes.bool,
};

export default Contacts;
