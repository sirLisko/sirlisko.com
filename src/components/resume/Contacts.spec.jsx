import React from "react";
import renderer from "react-test-renderer";

import Contacts from "./Contacts.jsx";

describe("Contacts Component", () => {
  let wrapper;

  it("should render properly", () => {
    wrapper = renderer.create(<Contacts />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render properly the alt version", () => {
    wrapper = renderer.create(<Contacts isAltVersion />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
