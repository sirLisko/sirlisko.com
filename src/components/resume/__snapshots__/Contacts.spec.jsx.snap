// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts Component should render properly 1`] = `
<article
  className="section contacts"
>
  <h1>
    Contacts
  </h1>
  <p>
    My website 
    <a
      href="https://sirlisko.com"
    >
      https://sirlisko.com
    </a>
  </p>
  <p>
    My Github
     
    <a
      href="https://github.com/sirlisko"
    >
      https://github.com/sirlisko
    </a>
  </p>
  <p>
    My email address 
    <a
      href="mailto:luca@sirlisko.com"
    >
      luca@sirlisko.com
    </a>
    .
  </p>
  <p>
    A print-friendly version of this:
     
    <a
      href="https://sirlisko.com/cv/alt.pdf"
    >
      https://sirlisko.com/cv/alt.pdf
    </a>
  </p>
</article>
`;

exports[`Contacts Component should render properly the alt version 1`] = `
<article
  className="section contacts"
>
  <h1>
    Contacts
  </h1>
  <p>
    My website 
    <a
      href="https://sirlisko.com"
    >
      https://sirlisko.com
    </a>
  </p>
  <p>
    My Github
     
    <a
      href="https://github.com/sirlisko"
    >
      https://github.com/sirlisko
    </a>
  </p>
  <p>
    My email address 
    <a
      href="mailto:luca@sirlisko.com"
    >
      luca@sirlisko.com
    </a>
    .
  </p>
  <p>
    A live version of this resume:
     
    <a
      href="https://sirlisko.com/resume"
    >
      https://sirlisko.com/resume
    </a>
  </p>
</article>
`;
